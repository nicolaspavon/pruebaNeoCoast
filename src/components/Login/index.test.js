import { render, fireEvent, screen } from '@testing-library/react';
import LoginForm from './index';

it('Should have login button disabled and show error tooltip when no username and password', () => {
  const { container } = render(<LoginForm />);
  const usernameInput = screen.getByRole('textbox', {
    name: 'Username',
  });
  const passwordInput = screen.getByRole('textbox', {
    name: 'Password',
  });
  const loginButton = screen.getByRole('button');
  expect(usernameInput).toBeInTheDocument();
  expect(passwordInput).toBeInTheDocument();
  fireEvent.change(usernameInput, { target: { value: '' } });
  fireEvent.change(passwordInput, { target: { value: '' } });
  expect(loginButton).toBeDisabled();
  expect(loginButton).toHaveTextContent('Login');
  expect(loginButton).toHaveAttribute('title', 'Username is missing. Password is missing. ');
  expect(container).toMatchSnapshot();
});

it('Should have login button disabled and show error tooltip when no username', () => {
  const { container } = render(<LoginForm />);
  const usernameInput = screen.getByRole('textbox', {
    name: 'Username',
  });
  const passwordInput = screen.getByRole('textbox', {
    name: 'Password',
  });
  const loginButton = screen.getByRole('button');
  expect(usernameInput).toBeInTheDocument();
  expect(passwordInput).toBeInTheDocument();
  fireEvent.change(usernameInput, { target: { value: '' } });
  fireEvent.change(passwordInput, { target: { value: 'contra' } });
  expect(loginButton).toBeDisabled();
  expect(loginButton).toHaveTextContent('Login');
  expect(loginButton).toHaveAttribute('title', 'Username is missing. ');
  expect(container).toMatchSnapshot();
});

it('Should have login button disabled and show error tooltip when no password', () => {
  const { container } = render(<LoginForm />);
  const usernameInput = screen.getByRole('textbox', {
    name: 'Username',
  });
  const passwordInput = screen.getByRole('textbox', {
    name: 'Password',
  });
  const loginButton = screen.getByRole('button');
  expect(usernameInput).toBeInTheDocument();
  expect(passwordInput).toBeInTheDocument();
  fireEvent.change(usernameInput, { target: { value: 'nico' } });
  fireEvent.change(passwordInput, { target: { value: '' } });
  expect(loginButton).toBeDisabled();
  expect(loginButton).toHaveTextContent('Login');
  expect(loginButton).toHaveAttribute('title', 'Password is missing. ');
  expect(container).toMatchSnapshot();
});

it('Should have login button enabled and show no tooltip', () => {
  const { container } = render(<LoginForm />);
  const usernameInput = screen.getByRole('textbox', {
    name: 'Username',
  });
  const passwordInput = screen.getByRole('textbox', {
    name: 'Password',
  });
  const loginButton = screen.getByRole('button');
  expect(usernameInput).toBeInTheDocument();
  expect(passwordInput).toBeInTheDocument();
  fireEvent.change(usernameInput, { target: { value: 'nico' } });
  fireEvent.change(passwordInput, { target: { value: 'pass' } });
  expect(loginButton).not.toBeDisabled();
  expect(loginButton).toHaveTextContent('Login');
  expect(loginButton).toHaveAttribute('title', '');
  expect(container).toMatchSnapshot();
});

it('Should change button text to loading', () => {
  const { container } = render(<LoginForm />);
  const usernameInput = screen.getByRole('textbox', {
    name: 'Username',
  });
  const passwordInput = screen.getByRole('textbox', {
    name: 'Password',
  });
  const loginButton = screen.getByRole('button');
  expect(usernameInput).toBeInTheDocument();
  expect(passwordInput).toBeInTheDocument();
  fireEvent.change(usernameInput, { target: { value: 'nico' } });
  fireEvent.change(passwordInput, { target: { value: 'pass' } });
  expect(loginButton).not.toBeDisabled();
  fireEvent.click(loginButton);
  expect(loginButton).toHaveTextContent('Loading');
  expect(container).toMatchSnapshot();
});

// Otros posibles tests a implementar:
// Se podr√≠a hacer un mock de la promise "doLogin", y chequear que el boton vuelva a decir login luego de que se cumple la promise