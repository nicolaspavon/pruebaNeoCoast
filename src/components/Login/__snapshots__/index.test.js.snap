// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should change button text to loading 1`] = `
<div>
  <form
    class="login_form"
  >
    <input
      aria-label="Username"
      placeholder="Username"
      type="text"
      value="nico"
    />
    <input
      aria-label="Password"
      placeholder="Password"
      role="textbox"
      type="password"
      value="pass"
    />
    <button
      disabled=""
      title=""
      type="submit"
    >
      Loading
    </button>
  </form>
</div>
`;

exports[`Should have login button disabled and show error tooltip when no password 1`] = `
<div>
  <form
    class="login_form"
  >
    <input
      aria-label="Username"
      placeholder="Username"
      type="text"
      value="nico"
    />
    <input
      aria-label="Password"
      placeholder="Password"
      role="textbox"
      type="password"
      value=""
    />
    <button
      disabled=""
      title="Password is missing. "
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`Should have login button disabled and show error tooltip when no username 1`] = `
<div>
  <form
    class="login_form"
  >
    <input
      aria-label="Username"
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      aria-label="Password"
      placeholder="Password"
      role="textbox"
      type="password"
      value="contra"
    />
    <button
      disabled=""
      title="Username is missing. "
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`Should have login button disabled and show error tooltip when no username and password 1`] = `
<div>
  <form
    class="login_form"
  >
    <input
      aria-label="Username"
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      aria-label="Password"
      placeholder="Password"
      role="textbox"
      type="password"
      value=""
    />
    <button
      disabled=""
      title="Username is missing. Password is missing. "
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;

exports[`Should have login button enabled and show no tooltip 1`] = `
<div>
  <form
    class="login_form"
  >
    <input
      aria-label="Username"
      placeholder="Username"
      type="text"
      value="nico"
    />
    <input
      aria-label="Password"
      placeholder="Password"
      role="textbox"
      type="password"
      value="pass"
    />
    <button
      title=""
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
